cyberbrain/tracer.py

```diff
@@ -156,6 +159,17 @@ def f():
         """
 
         def decorator(f, disabled_by_user=False):
+            stack_summary = traceback.extract_stack()
+            func_index = (
+                stack_summary.index(
```

Comment:
Can you extrac this part to a separate function, and add some comments? Also, though it seems to work, I'm not totally convinced that `FrameSummary.lineno` always produces the correct line no. Since you understand it better than me, some explanation in the comments would be appreciated.

Reply:
> 
> 
> Can you extract this part to a separate function, and add some comments?
> 
> Though it seems to work, I'm not totally convinced that `FrameSummary.lineno` always produces the correct line no. Since you understand it better than me, some explanation in the comments would be appreciated.

It's more consistent now because the trackback summary stack is only generated when the tracer's decorator is called. So from the python program's state, it should always be the tracer.py in the stack just after the decorated function. Then once the defined lineno is set. The traceback._active is set to False so that a tracer's don't share the same defined lineno. Only when the tracer decorator is called again will a defined lineno be assigned as the traceback active will be set to True. I hope that explanation makes sense.